robottelo/host_helpers/api_factory.py

```diff
@@ -628,7 +629,8 @@ def __init__(self, temp):
         :param entities.ProvisioningTemplate temp: entities.ProvisioningTemplate's object
         """
         self.temp = temp
-        if not isinstance(self.temp, self._satellite.api.ProvisioningTemplate):
+
+        if not isinstance(self.temp, entities.ProvisioningTemplate):
```

Comment:
is `self._satellite.api.ProvisioningTemplate` not working? why use entities?

Reply:
It's not working. So had to use entities here.

Reply:
what's an error? It should work, try adding `self.__dict__.update(initiate_repo_helpers(self._satellite))` to `__init__` of APIFactory class, as similar to CLIFactory

Reply:
This does work. Thanks @Gauravtalreja1 . 
@LadislavVasina1 Can you please add the suggested changes by Gaurav.

Reply:
@Gauravtalreja1 Why make it unnecessarily complicated? What is the goal of your suggestion, Gaurav?
I do not find anything wrong with keeping entities imported here.

Reply:
@ogajduse Yes, entities work too but just to keep it consistent with other helper classes as we using API property everywhere, and despite what I've discussed with @shweta83 regarding using context manager decorator instead of class for it herein  https://github.com/SatelliteQE/robottelo/pull/10796 where it can work without any issues

Reply:
@Gauravtalreja1 @ogajduse  After debugging and looking at the usage of this, I agree with @ogajduse  that we do not need the check for template type. I have removed the check and updated it to use context manager decorator instead of class.

Reply:
As I looked through #10796 I realise that my PR is no longer needed. I will close it then.

Reply:
Thanks @LadislavVasina1